### DISCUSSION TITLE
How to specify a default font for all components?

### DISCUSSION THREAD

**Discussion by ifeanyiisitor** [tamagui/tamagui#788] [Q&A]
Given the following config

```

const config = createTamagui({
  fonts: {
    body: robotoCondensedRegular,
  },
})
```

If I want to use the body font, I have to specify it like this

```
<Text fontFamily="$body">Foo</Text>
```

But what I can't figure out is how to make the body font be the default font for all components so that I don't have to manually specify it. I basically want to able to do

```
<Text>Foo</Text>
```

And have it use the `body` font specified in the config by default. How can I acieve this?

**Comment by manghuli** [comment]
avoid using `<Text>`, you should use `<Paragraph>` and and H tags instead. they take the config's body and heading fonts respectively :)

**Reply by NandoMB** [reply]
https://tamagui.dev/docs/components/text

"Note: Paragraph renders to a p tag on web, which can cause issues when you nest them during SSR. If you don't mind rendering to a span, use SizableText, otherwise, be careful when nesting items inside a Paragraph."

**Comment by NandoMB** [comment]
You can do this:
```tsx
import { Text, styled, GetProps } from '@tamagui/core';

export const StyledText = styled(Text, {
  name: 'StyledText',
  fontFamily: '$poppins'
})

export type StyledTextProps = GetProps<typeof StyledText>
```
or this:
```tsx
import { createTamagui, createFont } from '@tamagui/core';

const Poppins = createFont({
  family: 'Poppins',
  // ...
});

const config = createTamagui({
  // ...
  fonts: {
    heading: Poppins, // overrides the default heading font
    body: Poppins, // overrides the default body font
    poppins: Poppins // introduce a specific token for your new font
  },
  // ...
});

```
😃
